include/crab/domains/fixed_tvpi_domain.hpp

```diff
@@ -213,6 +213,29 @@ class fixed_tvpi_domain
     }
   }
 
+  boost::optional<number_t> is_var_constant(const variable_t &x) {
+    interval_t ival = m_base_absval.at(x);
+    return ival.singleton();
+  }
+
+  void rewrite_apply_var(arith_operation_t op, const variable_t &x,
+                         const variable_t &y, const variable_t &z,
+                         unsigned coefficient) {
+    assert(coefficient > 1);
+
+    if (boost::optional<number_t> n = is_var_constant(z)) {
+      rewrite_apply(op, x, y, *n, coefficient);
+    } else if (op == OP_ADDITION || OP_MULTIPLICATION) {
+      // Ignore case if op is subtraction and division
+      if (boost::optional<number_t> n = is_var_constant(y)) {
```

Comment:
you forgot `m_base_absval -= ghost_x;` in case of sub or div

Reply:
fixed